### DISCUSSION TITLE
Zoom rectangle outside model boundary + disabling axis "strip" zoom

### DISCUSSION THREAD

**Discussion by Filip-Esroe** [oxyplot/oxyplot#1957] [Q&A] [answered]
Hello I just had a two quick questions:
1) Is it possible to somehow draw the zoom rectangle outside of the model (such as beyond the axis) right now the zoom rectangle can only be drawn inside the boundary, and If I click above the plot model it does not draw even though there is plenty of space above it. Currently I have it set up so that the plot is inside a tabControl (for WPF) 
![image](https://user-images.githubusercontent.com/113527356/200366295-c3c4380d-41bb-4476-84b6-15b96865bd39.png)
Right now the yellow area is the only place where a zoom rectangle can be drawn, but for the ease of use I wanted to ask if its possible to make it draw outside (in the rest of the tab view) as well.

2) Is it possible to somehow disable the axis "strip" zoom? (as shown below)
![image](https://user-images.githubusercontent.com/113527356/200366778-f005bf5a-8fb1-439b-ba20-77c5afd650d7.png)

I tried using
`
MainPlot.ActualModel.Axis[0].IsZoomEnabled = false;
MainPlot.ActualModel.Axis[0].IsZoomEnabled = false;
`
Unfortunately this also disabled the standard rectangle zoom as well.

Thanks a lot for the help!! :)

**Comment by VisualMelon** [comment, marked as answer]
This should be possible - if a bit fiddly - with a custom `PlotController` and `ZoomRectangleManipulator`: the main problem is that `GetAxesFromPoint` is used to pick up the ages: it's what determines to only zoom the one axes, but if you're happy to bodge things a bit, you can ignore what it says and assign the axes manually in a derived `AssignAxes`.

Rough outline: 
 - implement custom `ZoomRectangleManipulator` that assigns its own axes
 - create a new plot controller, and unbind the old view commands you don't want
 - bind new commands that set up the `ZoomRectangleManipulator`

Everything you need can be found by following this binding https://github.com/oxyplot/oxyplot/blob/develop/Source/OxyPlot/PlotController/PlotController.cs#L23

<hr />

Testing on a local copy, substituting these lines in `PlotManipulator` seems to produce the desired behaviour (but will have other effects also, e.g. panning won't be per-axes, and will throw if there is no model attached), so the above should work OK.

    this.XAxis = xaxis ?? this.PlotView.ActualModel.DefaultXAxis;
    this.YAxis = yaxis ?? this.PlotView.ActualModel.DefaultYAxis;

(https://github.com/oxyplot/oxyplot/blob/develop/Source/OxyPlot/PlotController/Manipulators/PlotManipulator.cs#L109-L110)

**Reply by Filip-Esroe** [reply]
Thank you very much for the reply, it helps a lot!
How would I go about substituting into PlotManipulator from my code?
right now I have oxyplot as a NuGet package,

Thanks a lot for the help!

**Reply by VisualMelon** [reply]
You can find some simple examples here. showing how to change a command binding: https://github.com/oxyplot/oxyplot/blob/develop/Source/Examples/ExampleLibrary/Examples/PlotControllerExamples.cs

A trivial example that shows how to set the plot-controller for WPF is here: https://gist.github.com/VisualMelon/05aacd019da9e5162c055398b0abdb1c

Here are some non-trivial custom manipulators that don't do what you want already (the 'all' in the name means _all_ axes, for when a plot has more than 2 axes), but can probably be quite easily modified to do so (don't overlook the method at the end which is another example of re-fitting the `PlotController`. I can't remember when I wrote all of this stuff or for which version of Oxyplot, but hopefully it's useful. https://gist.github.com/VisualMelon/e1e68246486e62fa0f31340524944c39
 - for zooming, see `ZoomAllRectangleManipulator`, where you can probably get away with just disabling the check on line 145, which is the bit where it checks if you are moused over an axis rather than the plot area-
 - for panning, see `PanAllManipulator` and see the same check on line 313.

**Reply by Filip-Esroe** [reply]
Thank you very much for the help!
